packages/core/src/routes/admin-user.ts

```diff
@@ -17,6 +20,46 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
     return next();
   });
 
+  router.post(
+    '/users',
+    koaGuard({
+      body: object({
+        username: string().min(3),
+        password: string().min(6),
+        name: string().min(3),
+      }),
+    }),
+    async (ctx, next) => {
+      const { username, password, name } = ctx.guard.body;
+
+      assertThat(
+        !(await hasUser(username)),
```

Comment:
nit: prefer avoid inline await 

`const hasUser = await hasUser(username);`

Reply:
inline await is commonly used in our code, do we need to discuss that?

Reply:
LOG-1574